apps/nerves_hub_api/lib/nerves_hub_api_web/plugs/user.ex

```diff
@@ -2,13 +2,58 @@ defmodule NervesHubAPIWeb.Plugs.User do
   import Plug.Conn
 
   alias NervesHubWebCore.{Accounts, Certificate}
-  alias NervesHubWebCore.Accounts.User
 
   def init(opts) do
     opts
   end
 
   def call(conn, _opts) do
+    conn
+    |> authenticate_user()
+    |> case do
```

Comment:
Maybe it would be nicer not to pipe into a case statement here 

```suggestion
    case authenticate_user(conn) do
```